javascript/react/props-ii/props-in-getinitialstate-is-an-anti-pattern.md

```diff
@@ -29,37 +29,41 @@ Passing down props from the parent to generate state in `getInitialState` can le
 An example of a duplication of source of truth:
 
 ```javascript
-var Duplication = React.createClass({
-  getInitialState: function() {
+class Duplication extends React.Component {
+  getInitialState() {
```

Comment:
This whole insight isn't applicable anymore. We could just remove this insight or somehow mark it as legacy

Reply:
Hmmm maybe not the whole insight but this part for sure

Reply:
The same anti-pattern can be shown using a `constructor` instead of `getInitialState`.

Like this:

```jsx
class Duplication extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      nameWithQualifier: "Country " + this.props.country
    }
  }

  render() {
    return (
      <div>
        {this.state.nameWithQualifier}
      </div>
    );
  }
}
```

Reply:
I replaced the code block with the one suggested.